core/src/main/scala/org/apache/spark/sql/tispark/TiRowRDD.scala

```diff
@@ -41,6 +40,47 @@ class TiRowRDD(override val dagRequest: TiDAGRequest,
                @transient private val sparkSession: SparkSession)
     extends TiRDD(dagRequest, physicalId, tiConf, tableRef, session, sparkSession) {
 
+  protected val logger: Logger = log
+
+  override protected def getPartitions: Array[Partition] = {
+    val partitions = super.getPartitions
+
+    if (!dagRequest.getUseTiFlash)
+      return partitions
+
+    //TODO: the region cache logic need rewrite.
+    //https://github.com/pingcap/tispark/issues/1170
+    val regionMgr = session.getRegionManager
+    partitions.map(p => {
+      val tiPartition = p.asInstanceOf[TiPartition]
+      val tasks = tiPartition.tasks
+        .map(task => {
+          val learnerList = task.getRegion.getLearnerList
+          val learnerStore = learnerList
+            .collectFirst {
+              case peer =>
+                val store = regionMgr.getStoreById(peer.getStoreId)
+                if (store.getLabelsList
+                      .exists(
+                        label =>
+                          label.getKey == tiConf.getTiFlashLabelKey && label.getValue == tiConf.getTiFlashLabelValue
```

Comment:
are we going to hard-code the label config?

Reply:
Yes